src/pages/transaction-signing/hooks/use-submit-stx-transaction.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+import {
+  deserializeTransaction,
+  parsePrincipalString,
+  StacksTransaction,
+  TokenTransferPayload,
+} from '@stacks/transactions';
+import { PostRequests, setupHeystackEnv } from '@tests/mocks/heystack';
+import { ProviderWithWalletAndRequestToken } from '@tests/state-utils';
+import { useMakeStxTransfer } from '@pages/transaction-signing/hooks/use-make-stx-transfer';
+import { stxToMicroStx } from '@common/stacks-utils';
+import { useHandleSubmitTransaction } from '@pages/transaction-signing/hooks/use-submit-stx-transaction';
+
+const MEMO = 'hello world';
+const AMOUNT = 25; // STX
+const RECIPIENT = 'ST21FTC82CCKE0YH9SK5SJ1D4XEMRA069FKV0VJ8N';
+const EXPECTED_FEE = 180;
+const EXPECTED_AMOUNT_IN_USTX = stxToMicroStx(AMOUNT).toNumber();
+const EXPECTED_RECIPIENT = parsePrincipalString(RECIPIENT).address;
+
+function getCleanMemo(memo: string) {
+  // very strange bug in stacks.js?
+  // need to do this in explorer, too
+  // the memo ends up as "hello world\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+  return memo.replace(
+    '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
+    ''
```

Comment:
Yeah this is because the memo is padded with "empty" bytes.  It could be more/less depending on size of memo.

Reply:
ah makes sense